### PR TITLE
Add JAX-based T-Digest implementation and refactor related modules

### PR DESCRIPTION
**Pull request by snouatin-datastorm**
### Summary

This pull request introduces a JAX-based T-Digest implementation for scalable and efficient quantile computation. It includes changes to the `TDigestModule`, `TDigestNormalizer`, and associated test cases, focusing on performance improvements, consistency, and modularity.

### Changes

- Added `JaxTDigest`: a T-Digest implementation leveraging JAX.
- Refactored `TDigestModule` to replace host callbacks with a pure JAX implementation.
- Introduced a `JAX-based T-Digest Normalizer` and corresponding unit tests.
- Updated the normalizer and T-Digest implementation to use `float32` precision for consistency.
- Removed `fastdigest` dependency from `pyproject.toml`.
- Updated test configurations to align with new T-Digest precision and initialization changes.

### Refactors

- Replaced `JaxGraph` and `JaxHyperEdgeSet` usage with `Graph` and `HyperEdgeSet`.
- Updated test cases to use the new constructs aligning with `JaxBackend`.

### Tests

- Added test cases for masking, batch shape handling, inference mode, and changes in initialization.
- Adjusted `TDigestNormalizer` tests to reflect the new modular implementation.

### CODE REVIEW

File: src/energnn/model/normalizer/jax_tdigest.py
Comment (bdonon) [inline src/energnn/model/normalizer/jax_tdigest.py:1]: This should be moved to `tdigest_normalizer.py`, right?

File: src/energnn/model/normalizer/jax_tdigest.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+        res = jnp.interp(q_arr, mid_q, means, left=self.min_value, right=self.max_value)
+
+        res = jnp.where(self.mass == 0, jnp.nan, res)
+        res = jnp.where(q_arr <= 0.0, self.min_value, res)
+        res = jnp.where(q_arr >= 1.0, self.max_value, res)
+
+        return res
+
+    def cdf_vec(self, x: ArrayLike) -> FloatArray:
```
Comment (bdonon) [inline src/energnn/model/normalizer/jax_tdigest.py]: Where is it used?

File: src/energnn/model/normalizer/jax_tdigest.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+    K = jnp.array(max_centroids, dtype=jnp.float32)
+    K_int = internal_capacity
+
+    values = centroids[:, 0]
+    weights = centroids[:, 1]
+
+    order = jnp.argsort(values, stable=True)
+    sorted_m = values[order]
+    sorted_c = weights[order]
```
Comment (bdonon) [inline src/energnn/model/normalizer/jax_tdigest.py]: I'm not sure but isn't there a confusion between m (mass?) and c (centroid?) ?

File: src/energnn/model/normalizer/jax_tdigest.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    if w_arr.ndim == 0:
+        return jnp.full_like(x, float(w_arr), dtype=jnp.float32)
+
+    w_arr = jnp.ravel(w_arr)
+    if w_arr.shape[0] != x.shape[0]:
+        raise ValueError("w must be either a scalar or have the same length as x.")
+    return w_arr
+
+
+def _compress_sorted(
```
Comment (bdonon) [inline src/energnn/model/normalizer/jax_tdigest.py]: This function is super important and not self explanatory. A comprehensive docstring with a small example would be useful.

File: src/energnn/model/normalizer/jax_tdigest.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    cluster_ids = compute_cluster_ids(sorted_c, total_w)
+
+    new_c = jax.ops.segment_sum(sorted_c, cluster_ids, num_segments=K_int)
+    new_m_weighted = jax.ops.segment_sum(sorted_m * sorted_c, cluster_ids, num_segments=K_int)
+
+    new_m = jnp.where(
+        new_c > 0, new_m_weighted / jnp.maximum(new_c, jnp.array(1e-7, dtype=jnp.float32)), jnp.array(-1e30, dtype=jnp.float32)
+    )
+    new_m = jnp.maximum.accumulate(new_m)
+
```
Comment (bdonon) [inline src/energnn/model/normalizer/jax_tdigest.py]: I found these lines a bit hard to follow. Maybe some comments detailing the equations would be nice.

**Review by bdonon — CHANGES_REQUESTED** [review]
It lacks docstring to quickly understand the logic behind the largest methods. Super important for maintainability.
Apart from that, great job ! See also Issues 91 and 92 which could be included in this PR.

File: src/energnn/model/normalizer/jax_tdigest.py (hunk 1 above)
Comment (snouatin-datastorm) [reply src/energnn/model/normalizer/jax_tdigest.py]: After checking, it isn't used anywhere. I had implemented the JaxTDigest class to looks exactly like the one in fastdigest. I'd like to keep this method, but we can also remove it since it isn't currently being used.

**Review by bdonon — APPROVED** [review]
Perfect 👍
